src/robot/running/bodyrunner.py

```diff
@@ -596,9 +602,14 @@ def _run_finally(self, data, run):
 class WhileLimit:
```

Comment:
Should add 
```python
def __init__(self, on_limit_message=None):
    self.on_limit_message = on_limit_message
```
and then in child class `__init__`s use `super().__init__(on_limit_message)`.